membership-common/src/main/scala/com/gu/memsub/subsv2/reads/ChargeListReads.scala

```diff
@@ -182,11 +182,24 @@ object ChargeListReads {
 
   implicit def readSupporterPlusV2ChargeList: ChargeListReads[SupporterPlusCharges] = new ChargeListReads[SupporterPlusCharges] {
 
-    override def read(cat: PlanChargeMap, charges: List[ZuoraCharge]): ValidationNel[String, SupporterPlusCharges] = {
+    def getBillingPeriod(charges: List[ZuoraCharge]): ValidationNel[String, BillingPeriod] = {
+      val billingPeriods = charges.flatMap(_.billingPeriod).distinct
+      billingPeriods match {
+        case Nil => Validation.failureNel("No billing period found")
+        case b :: Nil => Validation.success[NonEmptyList[String], BillingPeriod](b.toBillingPeriod)
+        case _ => Validation.failureNel("Too many billing periods found")
+      }
+    }
+
+    def getPricingSummaries(charges: List[ZuoraCharge]): ValidationNel[String, List[PricingSummary]] = {
       val pricingSummaries = charges.map(_.pricing)
-      Validation.success[NonEmptyList[String], SupporterPlusCharges](
-        SupporterPlusCharges(Month, pricingSummaries),
-      )
+      Validation
+        .success[NonEmptyList[String], List[PricingSummary]](pricingSummaries)
+        .ensure("No pricing summaries found".wrapNel)(_.nonEmpty)
+    }
+
+    override def read(cat: PlanChargeMap, charges: List[ZuoraCharge]): ValidationNel[String, SupporterPlusCharges] = {
+      (getBillingPeriod(charges) |@| getPricingSummaries(charges)).apply(SupporterPlusCharges)
```

Comment:
@rupertbates How does the code stop non SupporterPlus subs successfully validating via this code path?

Reply:
I've switched the ordering in readPaidChargeList so that supporter plus is the last one attempted

```scala
implicit def readPaidChargeList: ChargeListReads[PaidChargeList] = new ChargeListReads[PaidChargeList] {
  def read(cat: PlanChargeMap, charges: List[ZuoraCharge]): ValidationNel[String, PaidChargeList] = {
    readPaperChargeList.read(cat, charges).map(identity[PaidChargeList]) orElse2
      readPaidCharge[Benefit, BillingPeriod](readAnyProduct, anyBpReads).read(cat, charges) orElse2
      readSupporterPlusV2ChargeList.read(cat, charges).map(identity[PaidChargeList])
  }.withTrace("readPaidChargeList")
}
```

I'm not 100% clear if it matters if we use the 'wrong' deserialiser if they're all producing a PaidChargeList